corehq/util/context_processors.py

```diff
@@ -138,6 +138,25 @@ def js_toggles(request):
     }
 
 
+def js_privileges(request):
+    if not getattr(request, 'couch_user', None):
+        return {}
+
+    domain = None
+    if getattr(request, 'project', None):
+        domain = request.project.name
+    elif getattr(request, 'domain', None):
+        domain = request.domain
+
+    if not domain:
+        return {}
+
+    plan_version = Subscription.get_subscribed_plan_by_domain(domain)
+    return {
+        'privileges': list(get_privileges(plan_version)),
```

Comment:
If we're worried about performance impact due to passing all privileges down, we can add a filter here to only pass those down that we care about in the JS code. When checking a new privilege in JS, we would then just need to update this filter as well. Other suggestions welcome.

Reply:
Always good to think about, but I'm not worried about this - looks like there are ~50 privileges, and we're already passing down ~200 feature flags. In general we pass an enormous amount of js around, this probably isn't the most valuable place to optimize.